src/main/java/com/redhat/devtools/lsp4ij/features/documentSymbol/LSPBreadcrumbsRefreshListener.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.documentSymbol;
+
+import com.intellij.codeInsight.daemon.DaemonCodeAnalyzer;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.startup.ProjectActivity;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import com.intellij.ui.components.breadcrumbs.Crumb;
+import com.intellij.xml.breadcrumbs.BreadcrumbListener;
+import com.intellij.xml.breadcrumbs.BreadcrumbsPanel;
+import com.redhat.devtools.lsp4ij.LSPIJUtils;
+import com.redhat.devtools.lsp4ij.LanguageServerWrapper;
+import com.redhat.devtools.lsp4ij.ServerStatus;
+import com.redhat.devtools.lsp4ij.lifecycle.LanguageServerLifecycleListener;
+import com.redhat.devtools.lsp4ij.lifecycle.LanguageServerLifecycleManager;
+import kotlin.Unit;
+import kotlin.coroutines.Continuation;
+import org.eclipse.lsp4j.jsonrpc.MessageConsumer;
+import org.eclipse.lsp4j.jsonrpc.messages.Message;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Refreshes the breadcrumbs and sticky lines for open editor tabs when a language server has started.
+ */
+public class LSPBreadcrumbsRefreshListener implements ProjectActivity, LanguageServerLifecycleListener {
+
+    private static final Key<Boolean> ADDED_BREADCRUMB_LISTENER = Key.create(LSPBreadcrumbsRefreshListener.class + ".ADDED_BREADCRUMB_LISTENER");
+    private static final Key<Boolean> NEEDS_RESTART = Key.create(LSPBreadcrumbsRefreshListener.class + ".NEEDS_RESTART");
+
+    @Override
+    @Nullable
+    public Object execute(@NotNull Project project, @NotNull Continuation<? super Unit> continuation) {
+        LanguageServerLifecycleManager.getInstance(project).addLanguageServerLifecycleListener(this);
+        return null;
+    }
+
+    @Override
+    public void handleStatusChanged(LanguageServerWrapper languageServer) {
```

Comment:
Hopefully the majority of this is pretty straightforward, but basically when a language server has been started, find all open files for which it supports breadcrumbs and force their editor to re-query breadcrumbs -- which only helps for breadcrumbs and doesn't automatically update sticky lines -- and denote the associated file as needing analysis restart -- which does update sticky lines based on the latest breadcrumbs.